src/pydash/strings.py

```diff
@@ -2052,14 +2130,42 @@ def words(text, pattern=None):
     .. versionchanged:: 3.2.0
         Improved matching for one character words.
     """
-    return reg_exp_js_match(text, pattern or RE_WORDS)
+    if pattern is None:
+        if has_unicode_word(text):
+            return reg_exp_js_match(text, RE_UNICODE_WORDS)
+        else:
+            return reg_exp_js_match(text, RE_ASCII_WORDS)
+
+    return reg_exp_js_match(text, pattern)
 
 
 #
 # Utility functions not a part of main API
 #
 
 
+def compounder(text):
+    """
+    Remove single quote before passing into words(), required by certain functions such as
+    kebab_case, camel_case, start_case etc.
+
+    to match Lodash-style outputs
+    """
+    # Reference: https://github.com/lodash/lodash/blob/2f79053d7bc7c9c9561a30dda202b3dcd2b72b90/camelCase.js#L25 # noqa
+    return words(re.sub("['\u2019]", "", to_string(text)))
+
+
+def has_unicode_word(text):
+    """
+    Check if the text contains unicode or requires more complex regex to handle.
+
+    Reference: https://github.com/lodash/lodash/blob/master/words.js#L3
+    """
+    text = pyd.to_string(text)
+    result = re.findall("[a-z][A-Z]|[A-Z]{2}[a-z]|[0-9][a-zA-Z]|[a-zA-Z][0-9]|[^a-zA-Z0-9 ]", text)
```

Comment:
Can define the regex string as a constant.

Reply:
I have replaced them with RE_UNICODE_WORDS variable in commit [9e8c7dc](https://github.com/dgilland/pydash/pull/159/commits/9e8c7dc29f8762632bd67cd5c7131f0e33e56194)